src/rules/property-case/index.js

```diff
@@ -0,0 +1,42 @@
+import {
+  cssWordIsVariable,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "property-case"
+
+export const messages = ruleMessages(ruleName, {
+  expected: (actual, expected) => `Expected "${actual}" to be "${expected}"`,
+})
+
+export default function (expectation) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: [
+        "lower",
+        "upper",
+      ],
+    })
+    if (!validOptions) { return }
+
+    root.walkDecls(decl => {
+      const prop = decl.prop
+      // Ignore sass interpolation and css variables
+      if (/#{.+}/.test(decl.toString()) || cssWordIsVariable(prop)) { return }
```

Comment:
Perhaps a good time to move the sass interpolation into a utility as we're using it in quite a few places. Perhaps `containsInterpolation(string)`?